transport/http/inbound.go

```diff
@@ -55,9 +55,10 @@ func Mux(pattern string, mux *http.ServeMux) InboundOption {
 // sharing this transport.
 func (t *Transport) NewInbound(addr string, opts ...InboundOption) *Inbound {
 	i := &Inbound{
-		once:   sync.Once(),
-		addr:   addr,
-		tracer: t.tracer,
+		once:      sync.Once(),
+		addr:      addr,
+		tracer:    t.tracer,
+		transport: t,
```

Comment:
I'd argue that it hurting readability is the exception case, not the rule, but id actually never thought of a linter, that's a really good idea